spec/requests/physical_servers_spec.rb

```diff
@@ -304,4 +304,66 @@
       end
     end
   end
+
+  describe "Physical Server refresh action" do
+    context "with an invalid id" do
+      it "fails to refresh a physical server" do
+        api_basic_authorize(action_identifier(:physical_servers, :refresh, :resource_actions, :post))
+
+        post(api_physical_server_url(nil, 999_999), :params => gen_request(:refresh))
+
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context "without an appropriate role" do
+      it "fails to refresh a physical server" do
+        ps = FactoryGirl.create(:physical_server)
+
```

Comment:
nit - can you remove this empty line? setting up the authorization and creating the factory are both part of the setup, they can be together. Same with similar instances below.